minifi_main/JsonSchema.cpp

```diff
@@ -0,0 +1,435 @@
+      "type": "object",
+      "required": ["name", "id", "class"],
+      "properties": {
+        "name": {"type": "string"},
+        "class": {"type": "string"},
+        "id": {"$ref": "#/$defs/uuid"}
+      }
+    }, )" + controller_services + R"(]},
+    "processor": {"allOf": [{
+      "type": "object",
+      "required": ["name", "id", "class", "scheduling strategy"],
+      "additionalProperties": false,
+      "properties": {
+        "name": {"type": "string"},
+        "id": {"$ref": "#/$defs/uuid"},
+        "class": {"type": "string"},
+        "max concurrent tasks": {"type": "integer", "default": 1},
+        "penalization period": {"$ref": "#/$defs/time"},
+        "yield period": {"$ref": "#/$defs/time"},
+        "run duration nanos": {"$ref": "#/$defs/time"},
+        "Properties": {},
+        "scheduling strategy": {"enum": ["EVENT_DRIVEN", "TIMER_DRIVEN", "CRON_DRIVEN"]},
+        "scheduling period": {},
+        "auto-terminated relationships list": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "uniqueItems": true
+        }
+      }}, {
+        "if": {"properties": {"scheduling strategy": {"const": "EVENT_DRIVEN"}}},
+        "then": {"properties": {"scheduling period": false}}
+      }, {
+        "if": {"properties": {"scheduling strategy": {"const": "TIMER_DRIVEN"}}},
+        "then": {"required": ["scheduling period"], "properties": {"scheduling period": {"$ref": "#/$defs/time"}}}
+      }, {
+        "if": {"properties": {"scheduling strategy": {"const": "CRON_DRIVEN"}}},
+        "then": {"required": ["scheduling period"], "properties": {"scheduling period": {"$ref": "#/$defs/cron_pattern"}}}
+      }, )" + processors + R"(]
+    },
+    "remote_process_group": {"allOf": [{
+      "type": "object",
+      "required": ["name", "id", "Input Ports"],
+      "properties": {
+        "name": {"type": "string"},
+        "id": {"$ref": "#/$defs/uuid"},
+        "url": {"type": "string"},
+        "yield period": {"$ref": "#/$defs/time"},
+        "timeout": {"$ref": "#/$defs/time"},
+        "local network interface": {"type": "string"},
+        "transport protocol": {"enum": ["HTTP", "RAW"]},
+        "Input Ports": {
+          "type": "array",
+          "items": {"$ref": "#/$defs/remote_port"}
+        },
+        "Output Ports": {
+          "type": "array",
+          "items": {"$ref": "#/$defs/remote_port"}
+        }
+      }
+    }, {
+      "if": {"properties": {"transport protocol": {"const": "HTTP"}}},
+      "then": {"properties": {
+        "proxy host": {"type": "string"},
+        "proxy user": {"type": "string"},
+        "proxy password": {"type": "string"},
+        "proxy port": {"type": "integer"}
+      }}
+    }]},
+    "connection": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["name", "id", "source name", "source id", "source relationship names", "destination name", "destination id"],
```

Comment:
`source name` and `destination name` are not really required

Reply:
indeed, they are only used as fallbacks in case of missing ids

Reply:
removed